prompting/organic/organic_scoring_prompting.py

```diff
@@ -89,21 +94,20 @@ async def _blacklist_fn(self, synapse: StreamPromptingSynapse) -> Tuple[bool, st
         return synapse.dendrite.hotkey != settings.ORGANIC_WHITELIST_HOTKEY, ""
 
     @override
-    async def _on_organic_entry(
-        self, synapse: StreamPromptingSynapse, metagraph: bt.metagraph, wallet: bt.wallet
-    ) -> StreamPromptingSynapse:
+    async def _on_organic_entry(self, synapse: StreamPromptingSynapse) -> StreamPromptingSynapse:
         """Organic query handle."""
         logger.info(f"[Organic] Received from {synapse.dendrite.hotkey}, IP: {synapse.dendrite.ip}")
 
+        # TODO: Query one of the top N incentive miners, the rest keep random.
         uids = list(self.get_random_uids())
         completions: dict[int, dict] = {}
         token_streamer = partial(
             self._stream_miner_response,
             synapse,
             uids,
             completions,
-            metagraph=metagraph,
-            wallet=wallet,
+            metagraph=self.metagraph,
+            wallet=self.wallet,
```

Comment:
this was changed to inherit from settings - so we're only initialising wallet & metagraph in the settings and then import from there

`
metagraph = settings.METAGRAPH
wallet = settings.WALLET
`

Reply:
Addressed

Reply:
It's based on the work you started. If we proceed with global vars for the whole project, there is no need to pass them inside the object:
```
        # self._organic_scoring = OrganicScoringPrompting(
        #     axon=self.axon,
        #     synth_dataset=SynthDatasetConversation(),
        #     trigger_frequency=settings.ORGANIC_TRIGGER_FREQUENCY,
        #     trigger_frequency_min=settings.ORGANIC_TRIGGER_FREQUENCY_MIN,
        #     trigger=settings.ORGANIC_TRIGGER,
        #     trigger_scaling_factor=settings.ORGANIC_SCALING_FACTOR,
        #     llm_pipeline=self.llm_pipeline,
        #     dendrite=self.dendrite,
        #     metagraph=settings.METAGRAPH,
        #     update_scores=self.update_scores,
        #     tokenizer=self.llm_pipeline.tokenizer,
        #     get_random_uids=lambda _: get_random_uids(self, k=settings.ORGANIC_SAMPLE_SIZE, exclude=[]),
        #     wallet=settings.WALLET,
        #     _lock=self._lock,
        # )
 ```
 
However it’s not very error-prone, since currently all global vars are mutable